src/components/resource/_fields/coverage-statement/coverageStatementDecorator.js

```diff
@@ -0,0 +1,20 @@
+import createCalculateDecorator from 'final-form-calculate';
+
+export default createCalculateDecorator(
+  {
+    field: 'hasCoverageStatement',
+    updates: {
+      coverageStatement: (hasCoverageStatement, { coverageStatement }) => {
```

Comment:
Please name incoming params as hasCoverageStatementValue and resetFieldsValues. Then extract coverageStatement inside function body. It will add the clarity to the code.